### DISCUSSION TITLE
Load only the current language resource

### DISCUSSION THREAD

**Discussion by phminella** [i18next/i18next#2207] [Q&A]
Hey!

My website has many languages, and I would like to load only the resource of the used language. I cannot do it for my life, not sure what is wrong or how to do it. You can find my config below. Now it is creating a chunk with all translations and that increases the size considerably. 

Thanks for your help in advance!

```
module.exports = {
  i18n: {
    defaultLocale: "en",
    locales: [
      "en",
      "es",
      "fil",
      "fr",
      "id",
      "it",
      "ja",
      "ko",
      "ms",
      "pt",
      "th",
      "vi",
      "zh_cn",
      "zh_tw",
    ],
    localeDetection: false,
  },
  localeStructure: "{{lng}}/{{ns}}",
  localeExtension: "json",
  reloadOnPrerender: true,
  defaultNS: "translation",
  trailingSlash: true,
  nonExplicitSupportedLngs: false,
  partialBundledLanguages: true,
  load: "currentOnly",
};
```

**Comment by jamuhl** [comment]
https://github.com/i18next/i18next-resources-to-backend might help...depending on use case choose some other backend https://www.i18next.com/overview/plugins-and-utils#backends (most prominent https://github.com/i18next/i18next-http-backend)

**Reply by phminella** [reply]
I apologize in advance for my complete ignorance. But I have a i18n.ts file with the config below, from the time I was using only react-i18next, and not next-i18next, but now that I am using next-i18next, that file feels completely useless, as I can remove everything, or delete it, and it seems to make no different. This next-i18next and react-i18next are a bit confusing to me. Where should I have that configuration to make it apply to my project? Right now it is in i18n/i18n.ts, but like I mentioned, it doesnt seem to have any effect. Do I need to trigger this somewhere in the project?
Thank you!

```
import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import backend from "i18next-http-backend";
i18n
  .use(backend)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    lng: "en",
    fallbackLng: "en",
    // resources: {},
    load: "currentOnly",
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
    partialBundledLanguages: true,
  });
export default i18n;
```

**Reply by jamuhl** [reply]
for both react-i18next and next-i18next there are existing examples...just look at them and tweak to your need

**Reply by phminella** [reply]
Thank you for give me some direction! I will take a look. I was already giving up.
